compiler/rustc_parse/src/lexer/diagnostics.rs

```diff
@@ -0,0 +1,130 @@
+use super::UnmatchedBrace;
+use rustc_ast::token::Delimiter;
+use rustc_data_structures::fx::FxHashMap;
+use rustc_errors::Diagnostic;
+use rustc_span::source_map::SourceMap;
+use rustc_span::Span;
+
+pub struct TokenTreeDiagInfo {
+    /// Stack of open delimiters and their spans. Used for error message.
+    pub open_braces: Vec<(Delimiter, Span)>,
+    pub unmatched_braces: Vec<UnmatchedBrace>,
+
+    /// Used only for error recovery when arriving to EOF with mismatched braces.
+    pub last_unclosed_found_span: Option<Span>,
+
+    /// Collect empty block spans that might have been auto-inserted by editors.
+    pub empty_block_spans: FxHashMap<Span, Delimiter>,
+
+    /// Collect the spans of braces (Open, Close). Used only
+    /// for detecting if blocks are empty and only braces.
+    pub matching_block_spans: Vec<(Span, Span)>,
+}
+
+impl Default for TokenTreeDiagInfo {
```

Comment:
`#[derive(Default)]` can be used here instead of implementing manually.

Reply:
Fixed